astropy/coordinates/earth.py

```diff
@@ -645,33 +645,39 @@ def get_gcrs_posvel(self, obstime):
             The GCRS velocity of the object
         """
         # GCRS position
-        gcrs_data = self._get_gcrs(obstime).data
+        gcrs_data = self.get_gcrs(obstime).data
         obsgeopos = gcrs_data.without_differentials()
         obsgeovel = gcrs_data.differentials['s'].to_cartesian()
         return obsgeopos, obsgeovel
 
-    def _gravitational_redshift(self, obstime):
+    def gravitational_redshift(self, obstime,
+                               bodies=('sun', 'jupiter', 'moon', 'earth')):
         """Return the gravitational redshift at this EarthLocation.
 
-        Calculates the gravitational redshift, of order 3 m/s, due to the Sun,
-        Jupiter, the Moon, and the Earth itself.
+        Calculates the gravitational redshift, of order 3 m/s, due to the
+        requested solar system bodies.
 
         Parameters
         ----------
         obstime : `~astropy.time.Time`
             The ``obstime`` to calculate the redshift at.
 
+        bodies : list
```

Comment:
Oh, good point.  I think it *is* useful to have this option, though.  And the way to support that is to also include the masses (will add a commit doing that, but can back both of them out if we really don't want them).

(My main motivation is that I think this is one of those cases where someone is more likely to see it's not possible and give up rather than ask for supporting it... and it's fairly straightforward to make it work)

Reply:
I'm happy to think we should have the other bodies, but then surely we should just provide the masses for other planets in the solar system!

Reply:
Maybe one way is to allow users to pass in a dict keyed by name with masses, but have a default that contains the masses of the main solar system bodies.

Reply:
I like @mhvk's suggestion of a dict, but can accept either approach.

Reply:
I like @mhvk suggestion here too, will update to do that.